src/Akka.Persistence.Sql.Hosting/HostingExtensions.cs

```diff
@@ -0,0 +1,252 @@
+
+namespace Akka.Persistence.Sql.Hosting
+{
+    public static class HostingExtensions
+    {
+        /// <summary>
+        ///     Adds Akka.Persistence.SqlServer support to this <see cref="ActorSystem"/>.
+        /// </summary>
+        /// <param name="builder">
+        ///     The builder instance being configured.
+        /// </param>
+        /// <param name="connectionString">
+        ///     Connection string used for database access.
+        /// </param>
+        /// <param name="autoInitialize">
+        ///     <para>
+        ///         Should the SQL store table be initialized automatically.
+        ///     </para>
+        ///     <i>Default</i>: <c>false</c>
+        /// </param>
+        /// <param name="providerName">
+        ///     <para>
+        ///         A string constant defining the database type to connect to, valid values are defined inside
+        ///         <see cref="LinqToDB.ProviderName"/> static class.
+        ///         Refer to the Members of <see cref="LinqToDB.ProviderName"/> for included providers.
+        ///     </para>
+        /// </param>
+        /// <param name="mode">
+        ///     <para>
+        ///         Determines which settings should be added by this method call.
+        ///     </para>
+        ///     <i>Default</i>: <see cref="PersistenceMode.Both"/>
+        /// </param>
+        /// <param name="schemaName">
+        ///     <para>
+        ///         SQL schema name to table corresponding with persistent journal.
+        ///     </para>
+        ///     <b>Default</b>: <c>null</c>
+        /// </param>
+        /// <param name="journalBuilder">
+        ///     <para>
+        ///         An <see cref="Action{T}"/> used to configure an <see cref="AkkaPersistenceJournalBuilder"/> instance.
+        ///     </para>
+        ///     <i>Default</i>: <c>null</c>
+        /// </param>
+        /// <param name="pluginIdentifier">
+        ///     <para>
+        ///         The configuration identifier for the plugins
+        ///     </para>
+        ///     <i>Default</i>: <c>"sql"</c>
+        /// </param>
+        /// <param name="isDefaultPlugin">
+        ///     <para>
+        ///         A <c>bool</c> flag to set the plugin as the default persistence plugin for the <see cref="ActorSystem"/>
+        ///     </para>
+        ///     <b>Default</b>: <c>true</c>
+        /// </param>
+        /// <returns>
+        ///     The same <see cref="AkkaConfigurationBuilder"/> instance originally passed in.
+        /// </returns>
+        /// <exception cref="ArgumentOutOfRangeException">
+        ///     Thrown when <see cref="journalBuilder"/> is set and <see cref="mode"/> is set to
+        ///     <see cref="PersistenceMode.SnapshotStore"/>
+        /// </exception>
+        public static AkkaConfigurationBuilder WithSqlServerPersistence(
+            this AkkaConfigurationBuilder builder,
+            string connectionString,
+            string providerName,
```

Comment:
Is it worthwhile to make this an enum rather than a magic string?

Reply:
Depends, we do not control these magic string constants, Linq2Db does.
We would need to remember to update the Enum based on changes in Linq2Db on each version update.